File: tutorial/cpp/stl/containers/queue/queue.cpp
// SPDX-License-Identifier: BSD-3-Clause

#include <iostream> // cout
#include <queue> // queue
using namespace std;

int main() {
    queue<int> q;
    for (int i = 0; i < 5; ++i) {
        q.push(i);
    }

    while (!q.empty()) {
        auto front = q.front();
        q.pop();
        cout << front << " "; // 0 1 2 3 4
    }
    cout << "\n";
    return 0;
}

File: tutorial/rust/std/containers/queue/queue.rs
// SPDX-License-Identifier: BSD-3-Clause
// Run: rustc queue.rs && ./queue

use std::collections::VecDeque;

fn main() {
    let mut q: VecDeque<i32> = VecDeque::new();
    for i in 0..5 {
        q.push_back(i);
    }

    while let Some(front) = q.pop_front() {
        print!("{} ", front);
    }
    println!();
    // print: 0 1 2 3 4
}
